src/webgpu/api/validation/vertex_state.spec.ts

```diff
@@ -396,7 +413,13 @@ g.test('vertex_shader_input_location_limit')
       },
     ];
 
-    const success = testLocation < kMaxVertexAttributes;
+    const success = {
+      // [[location(n)]] decoration must be a "non-negative i32 literal"
+      // TODO: Dawn raises a "Attribute location (N) over limits" error at
+      // shader creation time for N>15. Is this right?
```

Comment:
@Kangz - It seems to me that Dawn might be validating too early here. Please can you advise?

Reply:
https://github.com/gpuweb/gpuweb/pull/1423

Reply:
... restricted the locations themselves to be under the limit after some discussion that that was most likely sufficient, and forward-compatible with relaxing the restriction.

Reply:
Thanks. Made both the `shader` and `pipeline` conditions both dependent on the location being [0..kMaxVertexAttributes)